src/overlays/actors/ovl_Bg_Ddan_Jd/z_bg_ddan_jd.c

```diff
@@ -31,17 +33,152 @@ const ActorInit Bg_Ddan_Jd_InitVars = {
+        this->stepSpeed = SHORTCUT_STEP_SPEED;
+    } else {
+        this->stepSpeed = DEFAULT_STEP_SPEED;
+    }
+    this->actionFunc = BgDdanJd_Idle;
+}
+
+void BgDdanJd_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgDdanJd* this = THIS;
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+// Implements the platform's idle state
+void BgDdanJd_Idle(BgDdanJd* this, GlobalContext* globalCtx) {
+
+    if (this->idleTimer != 0) {
+        this->idleTimer--;
+    }
+
+    // if this is the platform that rises all the way to the top, and the switch state has turned on
+    if (this->stepSpeed == DEFAULT_STEP_SPEED && this->dyna.actor.params < 0x40 &&
+        Flags_GetSwitch(globalCtx, this->dyna.actor.params)) {
+        this->stepSpeed = SHORTCUT_STEP_SPEED;
+        this->state = STATE_GO_MIDDLE_FROM_BOTTOM;
+        this->idleTimer = 0;
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y + MOVE_HEIGHT_MIDDLE;
+        func_800800F8(globalCtx, 0xBF4, -0x63, &this->dyna.actor, 0);
+    }
+    if (this->idleTimer == 0) {
+        this->idleTimer = IDLE_FRAMES;
+        if (this->state == STATE_GO_BOTTOM) {
+            this->state = STATE_GO_MIDDLE_FROM_BOTTOM;
+            this->targetHeight = this->dyna.actor.initPosRot.pos.y + MOVE_HEIGHT_MIDDLE;
+        } else if (this->state == STATE_GO_MIDDLE_FROM_BOTTOM) {
+            // If platform goes all the way to the top
+            if (this->stepSpeed != DEFAULT_STEP_SPEED) {
+                this->state = STATE_GO_TOP;
+                this->targetHeight = this->dyna.actor.initPosRot.pos.y + MOVE_HEIGHT_TOP;
+            } else {
+                this->state = STATE_GO_BOTTOM;
+                this->targetHeight = this->dyna.actor.initPosRot.pos.y;
+            }
+        } else if (this->state == STATE_GO_MIDDLE_FROM_TOP) {
+            // If platform goes all the way to the top
+            if (this->stepSpeed != DEFAULT_STEP_SPEED) {
+                this->state = STATE_GO_TOP;
+                this->targetHeight = this->dyna.actor.initPosRot.pos.y + MOVE_HEIGHT_TOP;
+            } else {
+                this->state = STATE_GO_BOTTOM;
+                this->targetHeight = this->dyna.actor.initPosRot.pos.y;
+            }
+        } else if (this->state == STATE_GO_TOP) {
+            this->state = STATE_GO_MIDDLE_FROM_TOP;
+            this->targetHeight = this->dyna.actor.initPosRot.pos.y + MOVE_HEIGHT_MIDDLE;
+        }
+
+        this->actionFunc = BgDdanJd_Move;
+    }
+}
+
+// Handles dust particles and sfx when moving
+void BgDdanJd_MoveEffects(BgDdanJd* this, GlobalContext* globalCtx) {
+    Vec3f dustPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ddan_Jd/func_80870B88.s")
+    // Generate random dust particles at the platform's base.
+    dustPos.y = this->dyna.actor.initPosRot.pos.y;
+    if ((globalCtx->gameplayFrames & 1) != 0) {
```

Comment:
```suggestion
    if (globalCtx->gameplayFrames & 1) {
```

Reply:
i'll double check, but i think this change causes a non-matching